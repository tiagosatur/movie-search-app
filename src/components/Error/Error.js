import React from 'react';
import styled from 'styled-components';
import { colorPallete } from '../../style';

const Error = props => {
  const {
    message
  } = props;

  return (
    <div className='error'>
        <p>Oops...</p>
        <StyledError>{message}</StyledError>
    </div>

  );
};

const StyledError = styled.h3`
    color: ${colorPallete.light[33]};
    font-size: 24px;
`;

export default Error;
