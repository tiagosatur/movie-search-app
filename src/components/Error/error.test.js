import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Error from './Error';

describe('Error tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Error message='Error message' />
    );
  });
  
  it('should render the Error component', () => {
    expect(wrapper.find('.error').exists()).toBe(true);
  });

  it('should call Error snapshot correctly', () => {
    const tree = renderer
        .create(wrapper)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });
})