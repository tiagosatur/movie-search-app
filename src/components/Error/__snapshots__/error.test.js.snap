// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error tests should call Error snapshot correctly 1`] = `
<div
  className="error"
>
  <p>
    Oops...
  </p>
  <h3
    className="sc-bdVaJa eCtpUH"
  >
    Error message
  </h3>
</div>
`;
